core/src/main/scala/kafka/server/ZkAdminManager.scala

```diff
@@ -118,7 +118,7 @@ class ZkAdminManager(val config: KafkaConfig,
     metadataAndConfigs.get(topicName).foreach { result =>
       val logConfig = LogConfig.fromProps(LogConfig.extractLogConfigMap(config), configs)
       val createEntry = configHelper.createTopicConfigEntry(logConfig, configs, includeSynonyms = false, includeDocumentation = false)(_, _)
-      val topicConfigs = logConfig.nonInternalValues.asScala.map { case (k, v) =>
+      val topicConfigs = (logConfig.originals.asScala.filter(_._2 != null) ++ logConfig.nonInternalValues.asScala).map { case (k, v) =>
```

Comment:
Could we move ConfigHelper.describeConfigs.allConfigs to ConfigHelper and reuse it here?

Reply:
fixed